src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10600,13 +10600,31 @@ private ImplicitArrayCreationExpressionSyntax ParseImplicitlyTypedArrayCreation(
             var commas = _pool.Allocate();
             try
             {
-                while (this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                while (true)
```

Comment:
We had some parsing bugs with "while (true)" before.
Consider the new pattern and utility method:
```C#
                        int tokenProgress = -1;
                        while(IsMakingProgress(ref tokenProgress))
                        {
```

Reply:
will do.  thanks!